research-software-directory/RSD-as-a-service
# Isolate authentication tests

# Move postgrest row level authentication tests to separate docker compose

In order to run the test separately from the development and production I moved these tests under backend-postgrest/test folder and created separate auth-test.yml file to execute these test in separate docker environment. This way the the production or development database are not affected by running these tests.

How to test:

* navigate to backend-postgrest folder. The auth-test.yaml and test folder should be present
* See readme file in the test folder about how you can run test
* run test using this command from readme file 
```bash
# run test and stop containers
docker-compose -f auth-test.yml up \
  --abort-on-container-exit \
  --exit-code-from auth-test \
  && docker-compose -f auth-test.yml down --volumes
```
* all test should be completed with success, 46 test, 0 failed. All containers, volumes and the network should be removed like in the image below
![image](https://user-images.githubusercontent.com/9204081/149192620-b9f33ded-594d-48f0-ad0e-d8ab7e4efbb3.png)
